### ISSUE TITLE
Saving Instrument 'Navigation In the preset menu' is broken  [Linux]

### ISSUE THREAD

**Issue by mikobuntu** [zynaddsubfx/zyn-fusion-issues#48]
There is no way to edit the name for a newly created preset, the preset just takes on the name of the directory where you try to save. ( tested in both Renoise and Bitwig )
Bitwig engine will crash everytime I try to save a preset. I can probably upload some stack trace info if that will help?
thanks :)

**Comment by fundamental** [comment]
Hm, I know I fixed this crash once before, but I guess it's regressed again. If you can provide a stack trace that would be handy. I'm expecting it ends in readOnlyOp within Middleware.cpp

**Comment by fundamental** [comment]
I just tried the 3.0.1 build in renoise and bitwig on my slackware system. I couldn't get either to crash, nor could I get them to save a file to the incorrect location via the file menu. Would you mind elaborating the steps needed to get the crash? also, could you verify that you're running the 3.0.1 build?

**Comment by mikobuntu** [comment]
Just noticed that I'm running version 3.0.0 . I will update and see if I can replicate the crash. Since I have bought the plugin, what is my best option for updating the package? thanks :)

**Comment by fundamental** [comment]
Well, I was going to say just redownload from gumroad, but it looks like I didn't hit the "save changes" button there (why would they have that if there's already a multi-step process to upload new binaries...). Expect an email in a few minutes with new links.

**Comment by mikobuntu** [comment]
ok great job thanks :)

**Comment by fundamental** [comment]
Sent

**Comment by mikobuntu** [comment]
Ok all updated and Im getting the crash still with version 3.0.1 within Bitwig .
I also get the crash in standalone mode with console error 

[Zyn-Fusion:Error] Failed to read glpsol line...
value = -91423.250000
[DEBUG] Make Draw Sequence
[ERROR] Invalid parameter cannot be accessed at </file_list_dirs>
[ERROR] Invalid parameter cannot be accessed at </file_list_files>
[ERROR] Invalid parameter cannot be accessed at </file_home_dir>
[ERROR] Invalid parameter cannot be accessed at </config/favorites>
trace:
	[0] /home/mark/code/mruby-zest-build/src/mruby-widget-lib/mrblib/fcache.rb:36062:in Qml::FileSelector.updir
	[1] /home/mark/code/mruby-zest-build/src/mruby-widget-lib/mrblib/fcache.rb:36022:in Qml::FileSelector.set_file
	[2] /home/mark/code/mruby-zest-build/src/mruby-widget-lib/mrblib/fcache.rb:36292:in Qml::FileSelector.initialize
	[3] /home/mark/code/mruby-zest-build/src/mruby-widget-lib/mrblib/fcache.rb:9053:in Qml::SelColumn.cb
	[4] /home/mark/code/mruby-zest-build/src/mruby-widget-lib/mrblib/fcache.rb:9006:in Qml::SelColumn.onSetup
	[5] /home/mark/code/mruby-zest-build/src/mruby-widget-lib/mrblib/fcache.rb:1606:in Qml::Button.onMousePress
	[6] /home/mark/code/mruby-zest-build/src/mruby-widget-lib/mrblib/script.rb:218:in ZRunner.handleMousePress
	[7] /home/mark/code/mruby-zest-build/src/mruby-widget-lib/mrblib/script.rb:330:in ZRunner.handle_events
	[8] /home/mark/code/mruby-zest-build/mruby/mrblib/array.rb:17:in Array.each
	[9] /home/mark/code/mruby-zest-build/src/mruby-widget-lib/mrblib/script.rb:325:in ZRunner.handle_events
	[10] /home/mark/code/mruby-zest-build/src/mruby-widget-lib/mrblib/script.rb:582:in ZRunner.tick_events
/home/mark/code/mruby-zest-build/src/mruby-widget-lib/mrblib/fcache.rb:36062: undefined method 'reverse' for nil (NoMethodError)
[FATAL ERROR] Mruby Is Unable To Continue

**Comment by fundamental** [comment]
Ah, it's the file browser UI, not the stuff it's connected to. There's a few open issues on this component, so I'll try to fix a few of them before finishing a new linux 3.0.2 release candidate

**Comment by mikobuntu** [comment]
yes it seems that is the problem. I notice that it seems to be based on 'reverse' in the tree, where you click on a blank slot that takes you back one directory at a time, this will crash it everytime.

*edit*  I think the crash might actually be caused when we navigate back up the file tree ( when saving a preset ) and hit a wall at the root / of the file tree, where a call is being made to go up one more directory that doesn't actually exist. ( this only happens when clicking in the blank preset slots/ new unnamed slot )
![confirmedcrashzyn](https://cloud.githubusercontent.com/assets/335217/20738510/77af11dc-b6ab-11e6-9207-ed2d38545920.png)

A possible fix would be to disallow backward navigation ( if applicable? ) from the preset list, as we can do that in the preset/favorites folder menu.
